File: mater-rest/src/main/java/com/oneune/mater/rest/main/store/enums/RoleEnum.java
package com.oneune.mater.rest.main.store.enums;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.experimental.FieldDefaults;

/**
 * USER < SELLER < SUPPORT < ADMIN
 */
@FieldDefaults(level = AccessLevel.PRIVATE, makeFinal = true)
@AllArgsConstructor
@Getter
public enum RoleEnum {

    USER("Покупатель"),
    SELLER("Продавец"),
    SUPPORT("Поддержка"),
    ADMIN("Администратор");

    String rus;
}

File: mater-webapp/src/app/store/enums/role.enum.ts
export enum RoleEnum {
  USER = 'USER',
  SELLER = 'SELLER',
  SUPPORT = 'SUPPORT',
  ADMIN = 'ADMIN'
}

export const RoleTitle: { [key in RoleEnum]: string } = {
  [RoleEnum.USER]: 'Пользователь',
  [RoleEnum.SELLER]: 'Продавец',
  [RoleEnum.SUPPORT]: 'Поддержка',
  [RoleEnum.ADMIN]: 'Администратор'
}
